### PR TITLE
Take in multiple stitches such as "k5"

### PR DESCRIPTION
**Pull request by graceolivia**
Added to method in NestedArrayBuilder that takes something like `"k5"` and converts it to `"k1 k1 k1 k1 k1" `. This allows internal transformation is done right before the MetaDataBuilder stage where the pattern gets turned into an ASCII chart.

### CODE REVIEW

File: Sources/SwiftCliCore/NestedArrayBuilder.swift
Comment (breadoliveoilsalt) [inline Sources/SwiftCliCore/NestedArrayBuilder.swift:14]: Could be just `return row.flatMap...etc.`

File: Sources/SwiftCliCore/InputValidator.swift
Comment (hnlee) [inline Sources/SwiftCliCore/InputValidator.swift:35]: Consider refactoring this `inputValidation()` method for this PR. Suggestions:
- Use a verb name
- Split into the "three phases" you mentioned during pairing so that this method's responsibility is just to make sure the phases happen in the right order.

File: Tests/SwiftCliCoreTests/NestedArrayBuilderTest.swift
Comment (hnlee) [inline Tests/SwiftCliCoreTests/NestedArrayBuilderTest.swift:26]: Let's test some edge cases! What happens when you input "kx py"?

Comment (hnlee) [reply Tests/SwiftCliCoreTests/NestedArrayBuilderTest.swift:26]: As you said in pairing, let's also cover "k0"!

**Review by hnlee — CHANGES_REQUESTED** [review]

**Comment by hnlee** [comment]
Other ideas from pairing:
- Make `StitchLibrary` its own class.
- Think about... do we need a `Row` object (not just the metadata about the row, but an abstraction to represent the collection of stitches that is the row itself)?

File: Tests/SwiftCliCoreTests/NestedArrayBuilderTest.swift
Comment (graceolivia) [reply Tests/SwiftCliCoreTests/NestedArrayBuilderTest.swift:26]: Added these!

File: Sources/SwiftCliCore/NestedArrayBuilder.swift
Comment (graceolivia) [reply Sources/SwiftCliCore/NestedArrayBuilder.swift:14]: Done!

File: Sources/SwiftCliCore/InputValidator.swift
Comment (graceolivia) [reply Sources/SwiftCliCore/InputValidator.swift:35]: Changed the name and abstracted out the error handling! It feels more readable now. I also considered bundling up the error handling with the data transformation that happens with it at the same time (so a function would take `[String]`, run the `EmptyRowError` check, and then transform to `[[String]]`), but that seems to invalidate the single responsibility rule.  What do you think?

**Comment by graceolivia** [comment]
> Other ideas from pairing:
> 
> * Make `StitchLibrary` its own class.
> * Think about... do we need a `Row` object (not just the metadata about the row, but an abstraction to represent the collection of stitches that is the row itself)?

Not from this commit: If it's okay with you, I'm holding off on these for now because I need to resubmit my other PRs first, but I'll circle back once I have time.

File: Charts/.gitignore
Comment (hnlee) [inline Charts/.gitignore:3]: Hm, I would add these to the  `.gitignore` in the project root.

File: Sources/SwiftCliCore/InputValidator.swift
Comment (hnlee) [inline Sources/SwiftCliCore/InputValidator.swift:21]: Typo here.

File: Sources/SwiftCliCore/InputValidator.swift
Comment (hnlee) [inline Sources/SwiftCliCore/InputValidator.swift:63]: Is this a duplicate of `checkNoInvalidStitchesInNestedArray()`?

**Review by hnlee — CHANGES_REQUESTED** [review]
Just some clean up to do.

File: Charts/.gitignore
Comment (graceolivia) [reply Charts/.gitignore:3]: Okay! I needed to put something in there since otherwise git will ignore the directory, maybe I'll put a brief README instead.

Comment (breadoliveoilsalt) [reply Charts/.gitignore:3]: Grace, holler at me if you want to talk about this or have any questions.  (I'm a bit confused by git's behavior here.)

File: Sources/SwiftCliCore/Chartify.swift
Comment (breadoliveoilsalt) [inline Sources/SwiftCliCore/Chartify.swift:43]: Nice rename.

File: Sources/SwiftCliCore/InputValidator.swift
Comment (graceolivia) [reply Sources/SwiftCliCore/InputValidator.swift:63]: Oops! Thank you, deleted.

File: Sources/SwiftCliCore/NestedArrayBuilder.swift
Comment (breadoliveoilsalt) [inline Sources/SwiftCliCore/NestedArrayBuilder.swift:23]: Not a blocker -- something for pairing later after you merge -- let's talk about refactoring the names here and maybe making a little clearer.

File: Tests/SwiftCliCoreTests/NestedArrayBuilderTest.swift
Comment (breadoliveoilsalt) [inline Tests/SwiftCliCoreTests/NestedArrayBuilderTest.swift:36]: You can delete the extra lines above the expect.  Also, maybe play with leaving the code as is, but seeing if swift will allow you to break it into separate lines for readibility.  Like

`var obj = {name: "Tony", age: "Go away"}`
vs
```
var obj = {
  name: "Tony", 
  age: "Go away"
}
```

File: Charts/.gitignore
Comment (hnlee) [reply Charts/.gitignore]: FWIW, the `Charts/` directory ought to be created by the program if it doesn't exist, so I would actually just have `git` ignore the directory too.

**Comment by breadoliveoilsalt** [comment]
As of 10:30am Friday 11/5 -- expect no other comments from me!

File: Tests/SwiftCliCoreTests/NestedArrayBuilderTest.swift
Comment (graceolivia) [reply Tests/SwiftCliCoreTests/NestedArrayBuilderTest.swift:36]: That is more readable!

**Pull request merged**
